src/test/shell/bazel/cc_integration_test.sh

```diff
@@ -1382,4 +1382,29 @@ EOF
       fail "bazel build should've succeeded with --features=compiler_param_file"
 }
 
+function test_include_scanning_smoketest() {
+  cat > BUILD <<EOF
+cc_binary(
+  name = 'bin',
+  srcs = ['bin.cc'],
+  deps = [':spurious_dep'],
+)
+
+cc_library(
+  name = 'spurious_dep',
+  hdrs = ['dep.h'],
+)
+EOF
+
+  cat > bin.cc <<EOF
+#define NASTY "dep.h"
+#include NASTY
+int main() { return 0; }
+EOF
+
+  touch dep.h
+
+  bazel build --experimental_unsupported_and_brittle_include_scanning --features=cc_include_scanning //:bin && fail 'include scanning did not (wrongly) remove dependency' || true
```

Comment:
please add an assertion on the error message so the test doesn't break for some other reason